### PR TITLE
Add TimeUnixPTP to time formats

### PR DESCRIPTION
**Pull request by maxnoe**
### Description

This adds a new time format `TimeUnixPTP` that has epoch `1970-01-01T00:00:00 TAI` as apposed to `1970-01-01T00:00:08 TAI` of the `TimeUnixTAI` format.

This format is used by the Precision Time Protocol IEEE standard and by extension the White Rabbit hardware / software in high energy physics.

### CODE REVIEW

File: astropy/time/formats.py [hunk 1]
```diff
@@ -710,7 +710,7 @@ class TimeUnix(TimeFromEpoch):
 
 class TimeUnixTai(TimeUnix):
     """
-    Seconds from 1970-01-01 00:00:08 TAI (see notes), including leap seconds.
+    Counts the number of SI seconds elapsed since 1970-01-01 00:00:08 TAI.
```
Comment (taldcroft) [inline astropy/time/formats.py:713]: Please revert this unrelated change, the original is consistent in style with the other format class descriptions.

File: astropy/time/formats.py [hunk 2]
```diff
@@ -725,8 +725,9 @@ class TimeUnixTai(TimeUnix):
       ``1969-12-31 23:59:59.999918 UTC``.  The difference between TAI and UTC
       at that epoch was 8.000082 sec.
     - On the day of a leap second the difference between ``unix`` and ``unix_tai``
-      times increases linearly through the day by 1.0.  See also the
-      documentation for the `~astropy.time.TimeUnix` class.
+      times increases linearly through the day by 1.0.
```
Comment (taldcroft) [inline astropy/time/formats.py:728]: The original here was also fine, please revert.

File: astropy/time/formats.py [hunk 3]
```diff
@@ -745,6 +746,52 @@ class TimeUnixTai(TimeUnix):
     epoch_scale = 'tai'
 
 
+class TimeUnixPTP(TimeUnix):
+    """
+    Counts the number of SI seconds elapsed since 1970-01-01 00:00:00 TAI.
```
Comment (taldcroft) [inline astropy/time/formats.py:751]:
```suggestion
    Precision Time Protocol:  SI seconds elapsed from 1970-01-01 00:00:00 TAI.
```

File: astropy/time/formats.py [hunk 4]
```diff
@@ -745,6 +746,52 @@ class TimeUnixTai(TimeUnix):
...
+
+    This is basically the same as the ``unix_tai`` with an offset of 8 seconds
+    due to the different epoch (1970-01-01 00:00:00 UTC vs. TAI).
+
+    This will generally differ from Unix time by the cumulative integral number
+    of leap seconds since 1970-01-01 UTC plus the initial offset of 8 seconds.
+
+    This convention matches the definition of the Precision Time Protocol,
```
Comment (taldcroft) [inline astropy/time/formats.py:759]: Put this paragraph first in the docstring as the fundamental definition of the format, rather than starting with the format just being an 8-second derivative of `unix_tai`.

File: astropy/time/formats.py [hunk 5]
```diff
@@ -745,6 +746,52 @@ class TimeUnixTai(TimeUnix):
...
+    https://en.wikipedia.org/wiki/Precision_Time_Protocol and
+    also used by the White Rabbit protocol in High Energy Physics:
+    https://white-rabbit.web.cern.ch/
+
+    For a definition of the time scale, see:
```
Comment (taldcroft) [inline astropy/time/formats.py:764]: This is a bit confusing since one doesn't know the context of "the time scale", and this class is about a time format.  Maybe you mean "For more information about White Rabbit Based Clocks and Timestamps see https://www-acc.gsi.de/wiki/Timing/TimingSystemButisInterface#Clocks"?

File: CHANGES.rst [hunk 6]
```diff
@@ -68,6 +68,13 @@ astropy.tests
 astropy.time
 ^^^^^^^^^^^^
 
+- Add new time format ``unix_ptp``, that monotonically
+  counts number of seconds (no leap seconds are introduced)
+  since ``1970-01-01T00:00:00 TAI``.
+  This format is similar but different to ``unix_tai`` which uses an epoch
+  of ``1970-01-01T00:00:08 TAI`` and ``unix`` which uses an epoch of
+  ``1970-01-01T00:00:00 UTC`` and which is subject to leap seconds. [#11143]
```
Comment (taldcroft) [inline CHANGES.rst:76]:
```suggestion
- Add new time format ``unix_ptp`` (Precision Time Protocol) which provides 
  the number of SI seconds since ``1970-01-01T00:00:00 TAI``. [#11143]
```

**Review by taldcroft — CHANGES_REQUESTED** [review]
Looks basically good @maxnoe, but I have a number of documentation suggestions.

File: astropy/time/formats.py (hunk 1 above)
Comment (maxnoe) [reply astropy/time/formats.py:713]: The point is that the formulation `including leap seconds` is wrong.

Monotonic timescales do not have leap seconds, that's the point of them.

Leap seconds are a property of UTC.

File: astropy/time/formats.py (hunk 5 above)
Comment (maxnoe) [reply astropy/time/formats.py]: fixed

File: astropy/time/formats.py (hunk 4 above)
Comment (maxnoe) [reply astropy/time/formats.py:752]: done

File: astropy/time/formats.py (hunk 2 above)
Comment (maxnoe) [reply astropy/time/formats.py]: I reverted the unrelated docs changes and will open a second PR for those.

File: astropy/time/formats.py (hunk 1 above)
Comment (maxnoe) [reply astropy/time/formats.py]: I reverted the changes here and will add a second PR for those

**Review by taldcroft — APPROVED** [review]
Thanks @maxnoe!

**Comment by taldcroft** [comment]
@maxnoe - please see #11245. It turns out we miffed it with the original definition of `unix_tai`. The preferred plan now is to change `unix_tai` to be exactly what you implemented here for `unix_ptp`. 

Are you OK with backing out this PR in favor of just using `unix_tai`?

**Comment by maxnoe** [comment]
I am against changing the defintion of an already released time format. That will really confuse a lot of people I think.

**Comment by taldcroft** [comment]
Can you go over to #11245 and weigh in? The basic thought there is that `unix_tai` as defined was just wrong from the start and that likely nobody is really using it for that reason. (You are a perfect example.)  But you have a very valid point.

**Comment by maxnoe** [comment]
Sure. If it turns out that there is no place where the current definition of `unix_tai` is used, I am all for just calling it a bug and fixing it, at which point `unix_ptp` is just unnecessary.

**Comment by taldcroft** [comment]
It's not definitive, but there is not any application code on GitHub actually using `unix_tai` based searching GitHub for that term.

**Comment by maxnoe** [comment]
Ah, right.  That's indeed weird.
